frontend/apps/filemanager/filemanagerhistory.lua

```diff
@@ -134,6 +134,18 @@ function FileManagerHistory:onShowHist()
         onMenuHold = self.onMenuHold,
         _manager = self,
     }
+
+    -- Handle rotation events
+    local this = self
+    function self.hist_menu:onSetRotationMode(rotation)
+        if rotation ~= nil and rotation ~= Screen:getRotationMode() then
+            UIManager:close(this.hist_menu)
+            Screen:setRotationMode(rotation)
+            this:onShowHist()
```

Comment:
No way to make that a proper method of Menu ? It could access "this" via self._manager we set on line 135.

Reply:
Yeah, I vaguely thought about it but couldn't really find anything nicer than this approach, which is admittedly less than ideal :/.

Reply:
That did lead me to a possibly slightly less redundant approach in c4428de

And I also made it rotate the underlying widget, otherwise you're left with a broken layout there ;).

This means it's potentially slow in ReaderView, because it rotates the book. But it's potentially less weird than just inhibiting rotates inside ReaderView?

Reply:
And that's only actually slow when orientation lock is *disabled*, which, while the default, is generally not the setup gyro-fans use ;).